lib/src/utilities/yaru_carousel.dart

```diff
@@ -7,7 +7,7 @@ class YaruCarousel extends StatefulWidget {
     super.key,
     this.height = 500,
     this.width = 500,
-    required this.controller,
+    this.controller,
```

Comment:
i wanted to add `assert(controller == null || controller.pageLength == children.length)` but we'd have to make the constructor non-const... :(

Reply:
Another solution: remove `controller.pageLength` from the controller constructor, and fill it directly in `initState()` of `_YaruCarouselState`.
Like this we simplify the carousel instantiation, and we ensure that `controller.pageLength == children.length` while keeping a const constructor.

Reply:
Thanks, great idea! There seems to be a way to look up YaruCarousel from the scroll notification context (because YaruCarousel is a parent of the scrollable PageView) so we can get the correct value straight from the list of children. :)